Development/src/pages/devices/EditMatrixNamesField.js

```diff
@@ -0,0 +1,315 @@
+import { Button, IconButton, TextField, Tooltip } from '@material-ui/core';
+import get from 'lodash/get';
+import has from 'lodash/has';
+import CreateIcon from '@material-ui/icons/Create';
+import DeleteIcon from '@material-ui/icons/Delete';
+import ClearIcon from '@material-ui/icons/Clear';
+import DoneIcon from '@material-ui/icons/Done';
+import { isEmpty, set } from 'lodash';
+
+export const EditableIONameField = ({
+    defaultValue,
+    source,
+    label,
+    personalNames,
+    setPersonalNames,
+    autoFocus,
+    getLabelByLenght,
+    title,
+    ioKey,
+    ...props
+}) => {
+    const [displayRenameField, setDisplayRenameField] = useState(false);
+    const [displayEditIcon, setDisplayEditIcon] = useState(false);
+    const [value, setValue] = useState(() => {
+        if (get(personalNames, `${ioKey}.${source}.name`)) {
+            return get(personalNames, `${ioKey}.${source}.name`);
+        } else if (defaultValue != null) {
+            return defaultValue;
+        } else {
+            return '';
+        }
+    });
+
+    const inputRef = useRef();
+    useEffect(() => {
+        const timeout = setTimeout(() => {
+            if (autoFocus) inputRef.current.focus();
+        }, 100);
+        return () => {
+            clearTimeout(timeout);
+        };
+    }, [autoFocus]);
+
+    const getName = () => {
+        return get(personalNames, `${ioKey}.${source}.name`)
+            ? getLabelByLenght(get(personalNames, `${ioKey}.${source}.name`))
+            : getLabelByLenght(defaultValue);
+    };
+
+    const removeOverrideName = () => {
+        setDisplayRenameField(false);
+        setDisplayEditIcon(false);
+        setValue(defaultValue);
+        setPersonalNames(f => {
+            let newpersonalNames = { ...f };
+            if (has(newpersonalNames, `${ioKey}.${source}.name`)) {
+                set(newpersonalNames, `${ioKey}.${source}.name`, '');
+            }
+            if (isEmpty(get(newpersonalNames, `${ioKey}.${source}.channels`))) {
+                delete newpersonalNames[source];
+            }
+            return newpersonalNames;
+        });
+    };
+
+    const saveOverrideName = () => {
+        setPersonalNames(f => {
+            let newpersonalNames = { ...f };
+            if (!has(newpersonalNames, `${ioKey}`)) {
+                set(newpersonalNames, `${ioKey}`, {});
+            }
+            if (value === defaultValue) {
+            } else if (!has(newpersonalNames, `${ioKey}.${source}`)) {
+                set(newpersonalNames, `${ioKey}.${source}`, {
+                    name: value,
+                    channels: {},
+                });
+            } else {
+                set(newpersonalNames, `${ioKey}.${source}.name`, value);
+            }
+            return newpersonalNames;
+        });
+        setDisplayRenameField(false);
+        setDisplayEditIcon(false);
+    };
+
+    const cancleOverrideName = () => {
```

Comment:
`cancleOverrideName ` to `cancleEditName `

Reply:
Cancel not cancle 👍